test/suite/test_compact07.py

```diff
@@ -0,0 +1,222 @@
+            ds = SimpleDataSet(self, uri, self.table_numkv, 
+                            key_format=self.key_format, 
+                            value_format=self.value_format)
+            ds.populate()
+    
+            # Now let's delete a lot of data ranges. Create enough space so that compact runs in more
+            # than one iteration.
+            self.delete_range(uri)
+
+        # Reopen the connection to force the object to disk.
+        self.reopen_conn()
+        
+        small_file_free_space = self.get_free_space(uri_small)
+        
+        # Allow background compaction to run for some time. Set a low free_space_target larger
+        # than the smaller file.
+        self.session.compact(None,f'background=true,free_space_target={small_file_free_space + 1}MB')
+        
+        # Check that the background server is running
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        self.assertEqual(compact_running, 1)
+        stat_cursor.close()
+        
+        # Background compaction should run through every file as listed in the metadata file.
+        # Periodically check how many files we've compacted until we compact all of them.
+        while self.get_files_compacted() < self.n_tables:
+            time.sleep(2)
+            
+        # Check that we made no progress on the small file.
+        self.assertEqual(self.get_pages_rewritten(self.uri_prefix + '_small'), 0)
+        
+        # Check the background compaction server stats. We should have skipped at least once and 
+        # been successful at least once.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        skipped = stat_cursor[stat.conn.background_compact_skipped][2]
+        success = stat_cursor[stat.conn.background_compact_success][2]
+        self.assertGreater(skipped, 0)
+        self.assertGreater(success, 0)
+        stat_cursor.close()
+        
+        # Open a new session for foreground compaction, while leaving the background
+        # compaction server running.
+        session2 = self.conn.open_session()
+        # Use a free_space_target that is guaranteed to run on the small file.
+        session2.compact(uri_small,f'free_space_target={small_file_free_space - 1}MB')
+
+        time.sleep(1)
+        
+        # Check that foreground compaction has done some work on the small table.
+        self.assertGreater(self.get_pages_rewritten(uri_small), 0)
+                    
+        # Disable the background compaction server.
+        self.session.compact(None,'background=false')
+        
+        # Check that the background server is no longer running.
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        stat_cursor.close()
+        while compact_running:
```

Comment:
Instead of looping it continuously, how about sleeping for 1 sec and checking for the stat again?

Reply:
I agree, I added sleeps in 9739aa0659d74c9f7cf704f50050504bcc042e28.